agenta-web/src/hooks/useVaultSecret.ts

```diff
@@ -0,0 +1,147 @@
+import {useEffect, useRef, useState} from "react"
+import {
+    getAllProviderLlmKeys,
+    llmAvailableProviders,
+    llmAvailableProvidersToken,
+    LlmProvider,
+    removeSingleLlmProviderKey,
+    saveLlmProviderKey,
+} from "@/lib/helpers/llmProviders"
+import {isDemo} from "@/lib/helpers/utils"
+import {dynamicLib, dynamicService} from "@/lib/helpers/dynamic"
+
+export const useVaultSecret = () => {
+    const [secrets, setSecrets] = useState<LlmProvider[]>(llmAvailableProviders)
```

Comment:
might make sense to add a loading state. we can use this to improve ux on the pages / components using this hook, as there'll be a fetch each time this is mounted